### ISSUE TITLE
Sort by renews

### ISSUE THREAD

**Issue by oraclenut** [kyokan/bob-wallet#653]
Howdy, thank you for supporting this wallet. I have a lot of names and my issue is that trying to view which ones need to be renewed is painstaking difficult. Request that a "Renew All" or "Renew Oldest" or even sort by ones that need  to be renewed.
thank you

**Comment by d0peman** [comment]
I suggested bulk and automatic renews about 2 years ago at Handycon and it seemed to fall on deaf ears at the time.

I personally think it's a pretty crucial feature for a better overall user experience with Handshake in general. Especially if we want to move away from being tied to Namebase and truly decentralise name ownership and management going forward. I'd like to move away from Namebase and use bob-wallet but it's not something I want to have to struggle to use when managing a lot of names; I'd rather focus on developing other things.

It really would be nice to have an auto-renew toggle for each name as well as an option to set this for all names. Probably with a warning to the user that the mining fee will change with a link to information regarding this. I'd be happy to provide more feedback if needed

**Comment by rithvikvibhu** [comment]
I agree a better way to see expiring names would be nice. And being able to bulk renew a few selected names is coming as part of
- #631

Auto renews on the other hand is different and hasn't been done for a few reasons:
- Renew transactions can't just be signed beforehand and given to a watchtower (like you can technically do with scheduling reveals for bids). This is because a renew tx must include info from a "recent" block to prove that it wasn't signed months ago. That's also why it's not possible to presign and stack 10 years worth of renews at once.
- Signing a tx requires the wallet to be unlocked. Unlocking requires the password. So if auto renew is implemented in Bob, the wallet will need to stay unlocked or at least store the password somewhere, which isn't a great idea.
- Unlike naming systems based on smart contracts, it's not possible delegate some access to others - so we can't just say "let hs1... / kyokan renew this name on my behalf". Only the wallet owning the name can do renew. Maybe with https://github.com/handshake-org/hsd/pull/651 there might be something we can do, but it's still not ideal, all names would have to be on a script.
- And even after all this, if Bob adds an auto renew, it will need to stay open and synced all the time to be of any use. Which isn't what most people do. They check in every couple of days/weeks/months to use the wallet and close it. And if they are opening Bob, the renew option shows up for names expiring in the next 3 months. One-click renew is the closest feasible option right now.

So there's just a lot to consider for auto renewals in self-custody solutions (custody is easy since the service stays online and controls all the names).

**Comment by webelity** [comment]
Will #631 be merged into the next version? It would be an extremely handy feature.
